h2o-py/tests/testdir_algos/automl/pyunit_automl_distributions.py

```
@@ -0,0 +1,144 @@
+                                max_runtime_secs_per_model=1)
+                try:
+                    aml.train(y=scenario["response"], training_frame=df)
+                except Exception:
+                    assert scenario.get('fail', False), "This distribution should not have failed."
+                    return
+                assert not scenario.get('fail', False), "This distribution should have failed."
+                if aml.leaderboard.nrow == 0:
+                    algos = []
+                else:
+                    algos = list(set(get_leaderboard(aml, "algo").as_data_frame()["algo"].unique()))
+
+                for expected in ['DeepLearning', "DRF", 'GBM', 'GLM', 'StackedEnsemble', 'XGBoost']:
+                    assert expected in algos, "Expected {expected} but no found.".format(expected=expected)
+
+                for model_id in aml.leaderboard.as_data_frame()["model_id"]:
+                    distribution = get_distribution(model_id)
+
+                    assert distribution == expected_dist or \
+                    h2o.get_model(model_id).algo not in [a.lower() for a in scenario["algos"]], (
+                       "{model}: Expected distribution {s_dist} but {distribution} found!".format(
+                            model=model_id,
+                            s_dist=expected_dist,
+                            distribution=distribution
+                        ))
+            test_scenario.__name__ = "test_{}_distribution".format(distribution_name)
+            return test_scenario
+        tests.append(_(scenario))
+    return tests
+
+
+def test_python_api():
+    def test_parameterized_distribution_without_param():
+        try:
+            aml = H2OAutoML(distribution=dict(distribution="huber"))
+            aml = H2OAutoML(distribution="tweedie")
+            aml = H2OAutoML(distribution="quantile")
+        except AssertionError:
+            assert False, "should not have failed"
```

that's a strange assertion: why not let the error be propagated normally?